src/components/Web3Provider/index.tsx

```diff
@@ -15,7 +18,37 @@ export default function Web3Provider({ children }: { children: ReactNode }) {
 
   return (
     <Web3ReactProvider connectors={connectors} key={key}>
+      <Tracer />
       {children}
     </Web3ReactProvider>
   )
 }
+
+function Tracer() {
+  const { chainId, provider } = useWeb3React()
+  const networkProvider = RPC_PROVIDERS[(chainId || SupportedChainId.MAINNET) as SupportedChainId]
```

Comment:
think better not to default the chain here? may lead to weird sideeffects when switching chains unless i'm misunderstanding.

Reply:
but in that case wouldn't the chainId be passed in rather than using the chainId from web3-react? ex: i'm on Polygon, but use the Mainnet infura provider

Reply:
i'm pretty sure we want to just debug all the network providers instead of just the chainId we're on